app/controllers/users_controller.rb

```diff
@@ -0,0 +1,54 @@
+class UsersController < ApplicationController
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    user = User.new(user_params)
+    if user && user.save
+      session[:user_id] = user.id
+      redirect_to root_path
+    else
+      flash[:errors] = user.errors.full_messages
+      redirect_to new_user_path
+    end
+  end
+
+  def show
+    @user = User.find_by(id: current_user.id)
+    answers = Answer.where(user_id: current_user.id)
```

Comment:
If user has_many answers this line isn't needed. You can just directly user @user.answers